### ISSUE TITLE
Support arrays natively

### ISSUE THREAD

**Issue by iangow** [r-dbi/dbi3#44]
See [here](https://stat.ethz.ch/pipermail/r-sig-db/2015q1/001507.html) for some discussion we've had on R-SIG-DB on this issue in the past. 

[Here](http://stackoverflow.com/questions/30403546/how-do-i-turn-sets-into-indicators-of-set-membership) are hints of an indicative use case. Looking at `? tidyr::unnest`, it seems that what would be ideal would be if arrays could be mapped to what you call "list-columns" and then operations performed on those (in the example, I am doing the `unnest` in PostgreSQL).

**Comment by krlmlr** [comment]
Yes, perhaps in the very long term. What would the equivalent in Arrow look like?

**Comment by jjfantini** [comment]
+1
